### PR TITLE
Rtp 74 level bis

### PR DESCRIPTION
**Pull request by CyrilDehaese**

### CODE REVIEW

File: src/MapGeneration.cpp [hunk 1]
```diff
@@ -0,0 +1,492 @@
...
+                    else
+                        path = "assets/Blue Ground/TopBotLeftRight.png";
+                }
+            }
+        }
+        std::shared_ptr<Sprite> sprite = std::make_shared<Sprite>(path, 0.0f, 2.0f);
+        entity->addComponent(sprite);
+        std::shared_ptr<Position> position = std::make_shared<Position>(x * SCALE, (15 - y) * SCALE, 0);
+        entity->addComponent(position);
+        return entity;
```
Comment (antoinepnt) [inline src/MapGeneration.cpp:136]: You can use entity.addComponent().addComponent().etc() for better lisibility

File: src/MapGeneration.cpp [hunk 2]
```diff
@@ -0,0 +1,492 @@
...
+        file.open("map/Map Test.txt");
+        std::string lineOne = "****************";
+        std::string lineTwo;
+        std::string lineThree;
+        std::string strCube;
+        std::getline(file, lineTwo);
+        std::getline(file, lineThree);
+        for (int row = 0; row < 170 && lineThree != ""; row++)
+        {
+            if (lineTwo[0] == 'a')
```
Comment (antoinepnt) [inline src/MapGeneration.cpp:174]: STOP using magic values

File: src/MapGeneration.cpp [hunk 3]
```diff
@@ -0,0 +1,492 @@
...
+                    strCube.clear();
+                    strCube.push_back(lineTwo[line - 1]);
+                    strCube.push_back(lineOne[line]);
+                    strCube.push_back(lineThree[line]);
+                    strCube.push_back(lineTwo[line + 1]);
+                    scene->addEntity(whichEntity(strCube, row, line));
+                }
+                else if (lineTwo[line] == 'P')
+                {
+                    createPlayer(*scene, KEY_RIGHT, KEY_LEFT, KEY_UP, KEY_DOWN, KEY_RIGHT_CONTROL, 1, {row * SCALE, (15 - line) * SCALE, 0});
```
Comment (antoinepnt) [inline src/MapGeneration.cpp:208]: Remove "{}"

File: src/MapGeneration.cpp [hunk 4]
```diff
@@ -0,0 +1,492 @@
...
+            lineTwo = lineThree;
+            std::getline(file, lineThree);
+            if (lineThree == "" && lineTwo != "****************")
+                lineThree = "****************";
+        }
+        return scene;
+    }
+}
+
+// for (int row = 0; row < 170 && lineTwo != ""; row++)
```
Comment (antoinepnt) [inline src/MapGeneration.cpp:230]: Remove comment

**Review by antoinepnt — CHANGES_REQUESTED** [review]

**Review by iDommel — APPROVED** [review]

**Pull request merged**